EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -275,6 +277,14 @@ int RawDataUnpacker::ProcessVFATDataParallel(const uint16_t *buf, unsigned int O
 
   if (hFlag == vmRaw)
     wordsProcessed += 9;
+  
+  if (hFlag == vmDiamondCompact)
+  {
+    wordsProcessed--;
+    while ( (buf[wordsProcessed] & 0xFFF0)!= 0xF000 ) {
+      wordsProcessed++;
```

Comment:
should there be some run-away protection?
(same for another while loop in vmCluster)